src/app/left-menu/main/left-menu.component.ts

```diff
@@ -52,23 +53,112 @@ export class LeftMenuComponent implements OnDestroy, OnInit {
         e.stopPropagation();
     }
 
+    changeColor(text: string) {
+       const keyword = this.getRealName(this.searchKeyword.toLowerCase());
+       if (keyword.length >= 3) {
+        text = '<span style="color:#f05b41">' + text.slice(0, keyword.length) + '</span>' + text.slice(keyword.length);
+        return this.sanitizer.bypassSecurityTrustHtml(text);
+       }
+       return text;
+    }
+
     menuSearch() {
-        const keyword = this.searchKeyword.toLowerCase();
-        this.filteredMenuData = this.menuData.filter((value) => {
-            const name = value.name.toLowerCase();
-            const searchString = name + ', ' + value.equivalents;
-            return searchString.toLowerCase().indexOf(keyword) !== -1;
-        });
+        const keyword = this.getRealName(this.searchKeyword.toLowerCase());
+
+        if (keyword.length < 3) {
+            this.filteredData = [];
+            this.filteredData[0] = this.workArea;
+        }
+
+        if (keyword.length >= 3) {
+            this.filteredData = this.deepClone(this.menuData);
+
+            this.filteredData.forEach((data) => {
+                if (data.groups) {
+                    data.groups.forEach((group) => {
+                        if (group.items) {
+                            group.items = group.items.filter((value) => {
+                                const searchString = this.getRealName(value.Name);
+                                if(searchString.toLowerCase() === keyword) {
+                                    return true;
+                                } else if(searchString.toLowerCase().slice(0, keyword.length - searchString.length) === keyword) {
+                                    return true;
+                                } else {
+                                    return false;
+                                }
+                            });
+                        }
+                    });
+                } else {
+                    data.items = data.items.filter((value) => {
+                        const searchString = this.getRealName(value.Name);
+                        if (searchString.toLowerCase() === keyword) {
+                            return true;
+                        } else if (searchString.toLowerCase().slice(0, keyword.length - searchString.length) === keyword) {
+                            return true;
+                        } else {
+                            return false;
+                        }
+                    });
+                }
+            });
+
+            for(let i = 0; i < this.filteredData.length; i++) {
```

Comment:
This code is hard to understand. What do you want to do with these loops?

Reply:
I clear filtereddata from emty categories. If I don't this. view will look like this
![image](https://user-images.githubusercontent.com/52400194/63382260-0277c500-c3a3-11e9-89cb-632bfba91386.png)